### PR TITLE
Make build targets more debuggable

### PR DESCRIPTION
**Pull request by johnspackman**
This PR changes the uglify options so that the output is more easily debuggable, provided that the minify option is set to either "minify" or "beautify".  the default mode is "mangle" and that is unchanged.

### CODE REVIEW

**Comment by hkollmann** [comment]
Lint?

File: source/class/qx/tool/compiler/targets/BuildTarget.js [hunk 1]
```diff
@@ -174,13 +174,19 @@ module.exports = qx.Class.define("qx.tool.compiler.targets.BuildTarget", {
...
           break;
 
         case "beautify":
           uglifyOpts.mangle = false;
           uglifyOpts.output = {
             beautify: true
           };
+          uglifyOpts.compress = {
+              sequences: false
+          };
```
[the remark this answers is not in the document]
Comment (johnspackman) [reply source/class/qx/tool/compiler/targets/BuildTarget.js]: no - the idea is that mangle (maximum compress) doesn't worry about readability or debug-ability; if you want compression and debuggability, then you'd choose "minify", and if you want very little compression and maximum readbility go for "beautify".

[the remark this answers is not in the document]
Comment (derrell) [reply source/class/qx/tool/compiler/targets/BuildTarget.js]: I completely disagree. Code released to production should be mangled, but it should still be possible to debug it. These comma replacements are not necessary and should be disabled in all cases... or a switch should be added to to allow them to be disabled even in mangled code.

[the remark this answers is not in the document]
Comment (johnspackman) [reply source/class/qx/tool/compiler/targets/BuildTarget.js]: It is possible to debug it; you set a breakpoint on the first statement, and then you can move the breakpoint to the inner statements.

[the remark this answers is not in the document]
Comment (derrell) [reply source/class/qx/tool/compiler/targets/BuildTarget.js]: It's actually not that easy. With sourcemaps, you don't even see those commas unless you go explicitly looking for them, so don't know where you can and can't place breakpoints.

It seems you're resistant to having that "optimization" disabled in mangled mode. Can we at least have a switch that enables/disables it?

[the remark this answers is not in the document]
Comment (johnspackman) [reply source/class/qx/tool/compiler/targets/BuildTarget.js]: i suppose the space saving aren't really relevant, but i quite like having "mangle" mean that as much as possible  ... although in reality it's not brilliant for that purpose.  I should probably just try and find a proper mangler :)

**Comment by hkollmann** [comment]
@derrell : Is it ok for you now?

**Comment by derrell** [comment]
@hkollmann I don't think it has changed yet, but if I interpret @johnspackman 's comment correctly, he will make the change I requested to turn off sequences for mangle too.

**Comment by hkollmann** [comment]
He did with the last commit!

**Comment by derrell** [comment]
Indeed. My mistake.

**Comment by derrell** [comment]
Thanks, @johnspackman !

**Pull request merged**